File: motorcycle.py
from vehicle import Vehicle #
from garage import Garage 

class Motorcycle(Vehicle):

    # Membuat data atribut jenis motor, kapasitas tangki
    __jenismotor = ""
    __kapasitastangki = ""
    __garageObject = Garage("","","")

    # Constructor
    def __init__(self, plat_nomor , merk , tahun_produksi , warna , jenismotor , kapasitastangki , garage): # Membuat constructor dengan parameter
        super().__init__(plat_nomor, merk, tahun_produksi, warna, jenismotor, kapasitastangki) # Menginisialisasi setiap atribut yang ada di dalam konstruktor parent class Product
        # menginisialisasi masing masing atribut dengan nilai dari parameter nama atribut sesuai namanya
        self.__jenismotor = jenismotor
        self.__kapasitastangki = kapasitastangki
        self.__garageObject = garage

    # Getter dan Setter

    # Getter    
    def getJenisMotor(self):
        return self.__jenismotor
    
    def getKapasitasTangki(self):
        return self.__kapasitastangki
    
    def getGarage(self):
        return self.__garageObject

    
    # Setter
    def setJenisMotor(self, jenismotor):
        self.__jenismotor = jenismotor

    def setKapasitasTangki(self, kapasitastangki):
        self.__kapasitastangki = kapasitastangki

File: C++/Motorcycle.cpp
// Menggunakan pragma once agar header file hanya di-include satu kali
#pragma once
// Memasukan library yang digunakan, termasuk import file Induk
#include <iostream>
#include <string>
#include "Vehicle.cpp"

// Deklarasi namespace
using namespace std;

class Motorcycle : public Vehicle
{
private:
    /* data */
    string jenis_motor;
    string kapasitas_tangki;

public:
    /* Constructor */
    Motorcycle() : Vehicle() /* Membuat Constructor default tanpa isi parameter */
    {
        this->jenis_motor = "";
        this->kapasitas_tangki = "";
    }

    Motorcycle(string plat_nomor, string merk, int tahun_produksi, string warna, string jenis_motor, string kapasitas_tangki) : Vehicle(plat_nomor, merk, tahun_produksi, warna)
    {
        this->jenis_motor = jenis_motor;
        this->kapasitas_tangki = kapasitas_tangki;
    }

    /* Getter dan Setter */

    /* Getter */
    string getKapasitasTangki()
    {
        return this->kapasitas_tangki;
    }

    string getJenisMotor()
    {
        return this->jenis_motor;
    }

    /* Setter */
    void setKapasitasTangki(string kapasitas_tangki)
    {
        this->kapasitas_tangki = kapasitas_tangki;
    }

    void setJenisMotor(string jenis_motor)
    {
        this->jenis_motor = jenis_motor;
    }

    ~Motorcycle()
    {
    }
};
